hcache/compr_lz4.c

```diff
@@ -70,7 +70,7 @@ static void *compr_lz4_compress(void *cctx, const char *data, size_t dlen, size_
   /* int LZ4_compress_fast(const char* src, char* dst, int srcSize, int dstCapacity, int acceleration); */
   *clen = LZ4_compress_fast(data, cbuf + 4, datalen, len, C_HeaderCacheCompressLevel);
   if (*clen < 0)
```

Comment:
Can you please double check your expectations on return values? LZ4 [seems to return 0](https://github.com/inikep/lzbench/blob/master/lz4/lz4.h#L144) on failure, not some negative number.

Reply:
I was using "len" in the first time here ... don't remember why this was changed. I will fix this.